slack_bolt/slack_function/slack_function.py

```diff
@@ -78,8 +77,95 @@ def __call__(*args, **kwargs):
+
+            # Use this method as a decorator
+            @app.function("function_1")
+            def sample_view(event, complete: Complete):
+                client.views_open(
+                    interactivity_pointer=interactivity_pointer,
+                    view={...}
+                )
+
+            # Use this method as a decorator
+            @my_function.view("view_1")
+            def handle_submission(ack, body, client, view, complete):
+                # Acknowledge the view_submission event and close the modal
+                ack()
+                # Assume there's an input block with `block_c` as the block_id and `dreamy_input`
+                hopes_and_dreams = view["state"]["values"]["block_c"]["dreamy_input"]
+                user = body["user"]["id"]
+                # Validate the inputs
+                errors = {}
+                if hopes_and_dreams is not None and len(hopes_and_dreams) <= 5:
+                    errors["block_c"] = "The value must be longer than 5 characters"
+                if len(errors) > 0:
+                    ack(response_action="errors", errors=errors)
+                    return
+                # complete the function
+                complete()
+                # Do whatever you want with the input data - here we're saving it to a DB
+
+            # Pass a function to this method
+            sample_view_func = app.function("request-approval")(sample_view)
+            sample_view_func.action("view_1")(handle_submission)
 
-    @property
-    def __isabstractmethod__(self):
-        return getattr(self.func, "__isabstractmethod__", False)
+        To learn available arguments for middleware/listeners, see `slack_bolt.kwargs_injection.args`'s API document.
+        Args:
+            constraints: The conditions that match a request payload
+            matchers: A list of listener matcher functions.
+                Only when all the matchers return True, the listener function can be invoked.
+            middleware: A list of lister middleware functions.
+                Only when all the middleware call `next()` method, the listener function can be invoked.
+        """
+
+        def __call__(*args, **kwargs):
+            functions = extract_listener_callables(kwargs) if kwargs else list(args)
+            primary_matcher = builtin_matchers.function_view(self.callback_id, constraints, base_logger=self._base_logger)
+            return self._register_listener(list(functions), primary_matcher, matchers, middleware)
+
+        return __call__
+
+    def view_submission(
+        self,
+        constraints: Union[str, Pattern],
+        matchers: Optional[Sequence[Callable[..., bool]]] = None,
+        middleware: Optional[Sequence[Union[Callable, Middleware]]] = None,
+    ) -> Callable[..., Optional[Callable[..., Optional[BoltResponse]]]]:
+        """Registers a new `view_submission` listener."""
```

Comment:
Would devs use the `view` method above or these more specific `view_submission` / `view_closed` methods?

Reply:
Based on the [existing doc](https://slack.dev/bolt-python/concepts#view_submissions) for view_submission we recommend using `.view("my string")` to set up a `view_submission` listener, we don't have a dedicated section for `view_closed`

but here are some tested ways to set up `view_closed`
```python
app.view({"type": "view_closed", "callback_id": "view-id"})(simple_listener)
func.view_closed("view-id")(simple_listener)
```

Reply:
Ohhh that's right! Makes sense, and thanks for the explainer.